src/coord/src/catalog/config.rs

```diff
@@ -15,10 +15,7 @@ use build_info::BuildInfo;
 #[derive(Debug)]
 pub struct Config<'a> {
     /// The path to the catalog on disk.
-    ///
-    /// If set to `None`, indicates that an ephemeral in-memory catalog should
-    /// be used.
-    pub path: Option<&'a Path>,
+    pub path: &'a Path,
```

Comment:
How do tempfiles persist state when opening and reopening the catalog? (I would have guessed they would be destroyed when closed?)

Reply:
Because `path` takes a reference to some other value, that value must outlive the `Config`, i.e. it will be consistent as long as the `Config` is in use.